[Feature]: Please add support for pytest-8

### Suggested Behavior

Please consider adding support for pytest-8.

### Why is this needed?

The test suite currently fails with this version:

```
FAILED tests/unit/test_shortcuts.py::TestUnmarshalResponse::test_request_response_error - Failed: DID NOT WARN. No warnings of type (<class 'DeprecationWarning'>,) were emitted.
FAILED tests/unit/test_shortcuts.py::TestValidateRequest::test_spec_invalid - Failed: DID NOT WARN. No warnings of type (<class 'DeprecationWarning'>,) were emitted.
FAILED tests/unit/test_shortcuts.py::TestValidateRequest::test_spec_not_detected - Failed: DID NOT WARN. No warnings of type (<class 'DeprecationWarning'>,) were emitted.
FAILED tests/unit/test_shortcuts.py::TestValidateRequest::test_webhook_request_validator_not_found - Failed: DID NOT WARN. No warnings of type (<class 'DeprecationWarning'>,) were emitted.
FAILED tests/unit/test_shortcuts.py::TestValidateResponse::test_spec_not_detected - Failed: DID NOT WARN. No warnings of type (<class 'DeprecationWarning'>,) were emitted.
FAILED tests/unit/test_shortcuts.py::TestValidateResponse::test_spec_not_supported - Failed: DID NOT WARN. No warnings of type (<class 'DeprecationWarning'>,) were emitted.
FAILED tests/unit/test_shortcuts.py::TestValidateResponse::test_webhook_response_validator_not_found - Failed: DID NOT WARN. No warnings of type (<class 'DeprecationWarning'>,) were emitted.
```

Unless I'm mistaken, all the failures result from `pytest.warns()` being fixed to work inside `pytest.raises()`, and the warnings not actually being raised (previously, it was ignored).

### References

https://docs.pytest.org/en/stable/changelog.html#id38

### Would you like to implement a feature?

Yes

--- a/tests/unit/test_shortcuts.py
+++ b/tests/unit/test_shortcuts.py
@@ -423,8 +423,7 @@ class TestUnmarshalResponse:
         mock_unmarshal.return_value = ResultMock(error_to_raise=ValueError)
 
         with pytest.raises(ValueError):
-            with pytest.warns(DeprecationWarning):
-                unmarshal_response(request, response, spec=spec_v31)
+            unmarshal_response(request, response, spec=spec_v31)
 
         mock_unmarshal.assert_called_once_with(request, response)
 
@@ -597,15 +596,13 @@ class TestValidateRequest:
         request = mock.Mock(spec=Request)
 
         with pytest.raises(SpecError):
-            with pytest.warns(DeprecationWarning):
-                validate_request(request, spec=spec_invalid)
+            validate_request(request, spec=spec_invalid)
 
     def test_spec_not_detected(self, spec_v20):
         request = mock.Mock(spec=Request)
 
         with pytest.raises(SpecError):
-            with pytest.warns(DeprecationWarning):
-                validate_request(request, spec=spec_v20)
+            validate_request(request, spec=spec_v20)
 
     def test_request_type_invalid(self, spec_v31):
         request = mock.sentinel.request
@@ -733,8 +730,7 @@ class TestValidateRequest:
         request = mock.Mock(spec=WebhookRequest)
 
         with pytest.raises(SpecError):
-            with pytest.warns(DeprecationWarning):
-                validate_request(request, spec=spec_v30)
+            validate_request(request, spec=spec_v30)
 
     @mock.patch(
         "openapi_core.validation.request.validators.V31WebhookRequestValidator."
@@ -889,16 +885,14 @@ class TestValidateResponse:
         response = mock.Mock(spec=Response)
 
         with pytest.raises(SpecError):
-            with pytest.warns(DeprecationWarning):
-                validate_response(request, response, spec=spec_invalid)
+            validate_response(request, response, spec=spec_invalid)
 
     def test_spec_not_supported(self, spec_v20):
         request = mock.Mock(spec=Request)
         response = mock.Mock(spec=Response)
 
         with pytest.raises(SpecError):
-            with pytest.warns(DeprecationWarning):
-                validate_response(request, response, spec=spec_v20)
+            validate_response(request, response, spec=spec_v20)
 
     def test_request_type_invalid(self, spec_v31):
         request = mock.sentinel.request
@@ -965,8 +959,7 @@ class TestValidateResponse:
         response = mock.Mock(spec=Response)
 
         with pytest.raises(SpecError):
-            with pytest.warns(DeprecationWarning):
-                validate_response(request, response, spec=spec_v30)
+            validate_response(request, response, spec=spec_v30)
 
     @mock.patch(
         "openapi_core.validation.response.validators.V31WebhookResponseValidator."

